### PR TITLE
Nullable Types: infer_feature_types modifications

### PR DESCRIPTION
**Pull request by chukarsten**
Addresses: #3060 and #3061

Removing the ignore_nullable_types arg, the nullable types check from infer_feature_types and testing for infer_feature_types to raise an exception when they are detected.

### CODE REVIEW

File: evalml/data_checks/highly_null_data_check.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def validate(self, X, y=None):
         """
         results = {"warnings": [], "errors": [], "actions": []}
 
-        X = infer_feature_types(X, ignore_nullable_types=True)
+        X = infer_feature_types(X)
```
Comment (freddyaboulton) [inline evalml/data_checks/highly_null_data_check.py:108]: Does this mean that #3061 is also done?

**Review by bchen1116 — COMMENTED** [review]
Agreed with Freddy! I think this should be done in conjunction with #3061 so that we can ensure this change doesn't break anything for data checks

**Comment by chukarsten** [comment]
@bchen1116 @freddyaboulton I've added additional testing for the DefaultDataChecks suite as well as some of the data checks that explicitly deal with nulls to check that they jibe with the new nullable types.  Let me know what you think!

File: evalml/utils/woodwork_utils.py [hunk 2]
```diff
@@ -10,6 +10,8 @@
     ww.logical_types.Integer.type_string,
     ww.logical_types.Double.type_string,
     ww.logical_types.Boolean.type_string,
+    ww.logical_types.IntegerNullable.type_string,
```
Comment (freddyaboulton) [inline evalml/utils/woodwork_utils.py:13]: We should probably add Age/AgeNullable here? I wonder if we can get rid of this list though.

File: evalml/tests/data_checks_tests/test_data_checks.py [hunk 3]
```diff
@@ -216,7 +218,13 @@ def test_default_data_checks_classification(input_type):
 
     y = pd.Series([0, 1, np.nan, 1, 0])
     y_multiclass = pd.Series([0, 1, np.nan, 2, 0])
-    X.ww.init(logical_types={"natural_language_nan": "NaturalLanguage"})
+    X.ww.init(
+        logical_types={
```
Comment (freddyaboulton) [inline evalml/tests/data_checks_tests/test_data_checks.py:222]: Nice, I think this covering the default data checks gives us coverage for a lot of the data checks.

**Review by freddyaboulton — APPROVED** [review]
@chukarsten This looks good to me! Thanks for adding the coverage for the data checks + nullable types.

**Pull request merged**